controllers/osccluster_bastion_controller.go

```diff
@@ -0,0 +1,375 @@
+func checkBastionPublicIpOscAssociateResourceName(clusterScope *scope.ClusterScope) error {
+	var resourceNameList []string
+	clusterScope.Info("check match publicip with bastion")
+	bastionSpec := clusterScope.GetBastion()
+	bastionSpec.SetDefaultValue()
+	bastionPublicIpName := bastionSpec.PublicIpName + "-" + clusterScope.GetUID()
+	publicIpsSpec := clusterScope.GetPublicIp()
+	for _, publicIpSpec := range publicIpsSpec {
+		publicIpName := publicIpSpec.Name + "-" + clusterScope.GetUID()
+		resourceNameList = append(resourceNameList, publicIpName)
+	}
+	checkOscAssociate := Contains(resourceNameList, bastionPublicIpName)
+	if checkOscAssociate {
+		return nil
+	} else {
+		return fmt.Errorf("%s publicIp does not exist in bastion", bastionPublicIpName)
+	}
+}
+
+// checkBastionFormatParameters check Bastion parameters format.
+func checkBastionFormatParameters(clusterScope *scope.ClusterScope) (string, error) {
+	clusterScope.Info("Check Bastion parameters")
+	bastionSpec := clusterScope.GetBastion()
+	bastionSpec.SetDefaultValue()
+	bastionName := bastionSpec.Name + "-" + clusterScope.GetUID()
+	bastionTagName, err := tag.ValidateTagNameValue(bastionName)
+	if err != nil {
+		return bastionTagName, err
+	}
+
+	imageName := bastionSpec.ImageName
+	if imageName != "" {
+		_, err := infrastructurev1beta1.ValidateImageName(imageName)
+		if err != nil {
+			return bastionTagName, err
+		}
+	} else {
+		_, err := infrastructurev1beta1.ValidateImageId(bastionSpec.ImageId)
+		if err != nil {
+			return bastionTagName, err
+		}
+	}
+
+	bastionKeypairName := bastionSpec.KeypairName
+	_, err = infrastructurev1beta1.ValidateKeypairName(bastionKeypairName)
+	if err != nil {
+		return bastionTagName, err
+	}
+
+	vmType := bastionSpec.VmType
+	_, err = infrastructurev1beta1.ValidateVmType(vmType)
+	if err != nil {
+		return bastionTagName, err
+	}
+
+	bastionDeviceName := bastionSpec.DeviceName
+	_, err = infrastructurev1beta1.ValidateDeviceName(bastionDeviceName)
+	if err != nil {
+		return bastionTagName, err
+	}
+
+	bastionSubregionName := bastionSpec.SubregionName
+	_, err = infrastructurev1beta1.ValidateSubregionName(bastionSubregionName)
+	if err != nil {
+		return bastionTagName, err
+	}
+
+	bastionSubnetName := bastionSpec.SubnetName
+	clusterScope.Info("Get bastionSubnetName", "bastionSubnetName", bastionSubnetName)
+
+	ipSubnetRange := clusterScope.GetIpSubnetRange(bastionSubnetName)
+	bastionPrivateIps := clusterScope.GetBastionPrivateIps()
+	var subnetsSpec []*infrastructurev1beta1.OscSubnet
+	networkSpec := clusterScope.GetNetwork()
+	networkSpec.SetSubnetDefaultValue()
+	subnetsSpec = networkSpec.Subnets
+	for _, subnetSpec := range subnetsSpec {
+		subnetName := subnetSpec.Name
+		ipSubnetRange := subnetSpec.IpSubnetRange
+		clusterScope.Info("Get IpSubnetRange", "ipSubnetRange", ipSubnetRange)
+		clusterScope.Info("Get subnetName", "subnetName", subnetName)
+	}
```

Comment:
What is the use of this for loop ? 
Does it matter if `ipSubnetRange` is overrided if there are multiple subnets ?

Reply:
No need i will delete it.

Reply:
No it is now remove